code/io/joy_ff-sdl.cpp

```diff
@@ -86,11 +86,8 @@ int joy_ff_init()
 		}
 	}
 #endif
-	
-	// This should be how we do it, but it's currently bugged in SDL so it fails to re-enumerate haptic axes.
-	//haptic = SDL_HapticOpenFromJoystick(joy_get_device());
 
-	haptic = SDL_HapticOpen(0);
+	haptic = SDL_HapticOpenFromJoystick(io::joystick::getCurrentJoystick()->getDevice());
```

Comment:
A check for `currentJoystick` being `nullptr` should be added, or this line can cause a crash.